pages/api/books/POST.ts

```diff
@@ -48,27 +67,49 @@ export default async function postHandler(req: NextApiRequest) {
         'Invalid book, please check the book object and resend after correcting the data',
     };
   }
+  const { title, author, genres, regions, location } = body;
+  const getLocation = await prisma.location.findFirst({
+    where: { name: 'PEPO Checkin' },
+  });
 
-  const { title, author, genres, regions } = body;
+  // create genres and regions if they do not exist
```

Comment:
This was necessary in place of prisma's connectOrCreate command because prisma would keep deleting and recreating existing genres and regions.